components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/JITProvisioningPostAuthenticationHandler.java

```diff
@@ -1159,4 +1246,39 @@ private String getUserStoreDomain(String provisioningUserStoreId, UserRealm real
         }
         return userStoreDomain;
     }
+
+    private String getRoleClaimMapping(AuthenticationContext context) throws FrameworkException {
+
+        String spStandardDialect = (String) context.getProperty(FrameworkConstants.SP_STANDARD_DIALECT);
+        Map<String, String> spClaimMappings = context.getSequenceConfig().getApplicationConfig().
+                getClaimMappings();
+        Map<String, String> localToSPClaimMappings = null;
+
+        if (spStandardDialect != null) {
+            try {
+                localToSPClaimMappings = ClaimMetadataHandler.getInstance()
+                        .getMappingsMapFromOtherDialectToCarbon(spStandardDialect, null, context.getTenantDomain(),
+                                true);
+            } catch (ClaimMetadataException e) {
+                throw new FrameworkException("Error occurred while getting claim mappings from " +
+                        spStandardDialect + " dialect to " +
+                        ApplicationConstants.LOCAL_IDP_DEFAULT_CLAIM_DIALECT + " dialect for " +
+                        context.getTenantDomain() + " to handle federated claims", e);
+            }
+        } else if (!spClaimMappings.isEmpty()) {
+            localToSPClaimMappings = FrameworkUtils.getLocalToSPClaimMappings(spClaimMappings);
+        } else {
+            // no standard dialect and no custom claim mappings
+            throw new AssertionError("Authenticator Error! Authenticator does not have a " +
```

Comment:
Why this is an AssertionError instead of FrameworkException ?

Reply:
Fixed with